src/exhaustive-switch.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Here the compiler checks that we have handled all possible cases in the switch statement
+ * in other words, a compile-time exhaustive check
+ */
+
+/**
+ * This function enables the exhaustive check.
+ * It works by returning `never`, and accepting the first param as `never`
+ * @param _type The type that we're checking
+ * @param message A message to display if it errors
+ * @param lastWords A callback to call before the error
+ */
+export function assertUnreachable(
+  _type: never,
```

Comment:
I would only change this `_type` to `value` to not confuse the `type` and `value` spaces.

```suggestion
  value: never,
```